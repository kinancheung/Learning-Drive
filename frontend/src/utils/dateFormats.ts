// Date format constant
export const dateFormats = {
  post: 'h:mm a d/MM/yyyy',
};
