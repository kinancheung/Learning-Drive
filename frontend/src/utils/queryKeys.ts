import { postsSearchFilter } from './schema/searchPostsSchema';

export const queryKey = {
  CATEGORIES: 'categories',
  POSTS: (page: number, filter: postsSearchFilter): [string, number, postsSearchFilter] => [
    'posts',
    page,
    filter,
  ],
  PROFILE: (userId?: string): [string, string] => {
    if (!userId) throw new Error();
    return ['profile', userId];
  },
  POST: (postId: string): [string, string] => ['post', postId],
};
