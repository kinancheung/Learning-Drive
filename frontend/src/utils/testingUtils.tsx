import { SnackbarProvider } from 'notistack';
import { PropsWithChildren } from 'react';
import { HelmetProvider } from 'react-helmet-async';
import { MemoryRouter, Location } from 'react-router-dom';

import { AuthContext, AuthContextState } from '../context/AuthContext';

interface composeTestingWrapperOpts {
  initialEntries?: (string | Partial<Location>)[];
  initialIndex?: number;
  authState?: Partial<AuthContextState>;
}

export const composeTestingWrapper = ({
  initialEntries,
  initialIndex,
  authState,
}: composeTestingWrapperOpts = {}) => {
  const Wrapper = ({ children }: PropsWithChildren<{}>) => {
    const defaultState: AuthContextState = {
      signInWithGithub: jest.fn(),
      signInWithGoogle: jest.fn(),
      signInWithFacebook: jest.fn(),
      signOut: jest.fn(),
      ...authState,
    };

    return (
      <HelmetProvider>
        <MemoryRouter initialEntries={initialEntries} initialIndex={initialIndex}>
          <AuthContext.Provider value={defaultState}>
            <SnackbarProvider>{children}</SnackbarProvider>
          </AuthContext.Provider>
        </MemoryRouter>
      </HelmetProvider>
    );
  };
  return Wrapper;
};
