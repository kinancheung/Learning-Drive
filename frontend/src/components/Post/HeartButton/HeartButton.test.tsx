import { render } from '@testing-library/react';

import { HeartButton } from '.';

import { composeTestingWrapper } from '../../../utils/testingUtils';

describe('HeartButton', () => {
  it('renders without error', () => {
    const view = render(<HeartButton />, { wrapper: composeTestingWrapper() });

    expect(view.asFragment()).toMatchSnapshot();
  });

  it('renders filled without error', () => {
    const view = render(<HeartButton filled />, { wrapper: composeTestingWrapper() });

    expect(view.asFragment()).toMatchSnapshot();
  });
});
