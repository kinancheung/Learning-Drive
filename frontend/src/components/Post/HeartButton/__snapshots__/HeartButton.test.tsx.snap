// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeartButton renders filled without error 1`] = `
<DocumentFragment>
  <button
    aria-label="Unlike Post"
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
    data-mui-internal-clone-element="true"
    tabindex="0"
    type="button"
  >
    <div
      style="overflow: hidden; outline: none; width: 28px; height: 28px;"
    >
      <svg
        height="32"
        id="heart_i5001"
        preserveAspectRatio="xMidYMid meet"
        style="width: 100%; height: 100%; transform: translate3d(0,0,0);"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs>
          <clippath
            id="__lottie_element_10"
          >
            <rect
              height="32"
              width="32"
              x="0"
              y="0"
            />
          </clippath>
        </defs>
        <g
          clip-path="url(#__lottie_element_10)"
        >
          <g
            opacity="1"
            style="display: block;"
            transform="matrix(1,0,0,1,4,4)"
          >
            <g
              opacity="0"
              transform="matrix(0.5,0,0,0.5,12,11.845999717712402)"
            >
              <path
                d=" M8.84000015258789,-7.236000061035156 C6.692999839782715,-9.383999824523926 3.2090001106262207,-9.385000228881836 1.0609999895095825,-7.236999988555908 C1.0609999895095825,-7.236999988555908 1.059999942779541,-7.236000061035156 1.059000015258789,-7.236000061035156 C1.059000015258789,-7.236000061035156 -0.0010000000474974513,-6.176000118255615 -0.0010000000474974513,-6.176000118255615 C-0.0010000000474974513,-6.176000118255615 -1.0609999895095825,-7.236000061035156 -1.0609999895095825,-7.236000061035156 C-3.2090001106262207,-9.383999824523926 -6.690999984741211,-9.383999824523926 -8.84000015258789,-7.236000061035156 C-10.98799991607666,-5.0879998207092285 -10.98799991607666,-1.6039999723434448 -8.84000015258789,0.5440000295639038 C-8.84000015258789,0.5440000295639038 -7.781000137329102,1.6039999723434448 -7.781000137329102,1.6039999723434448 C-7.781000137329102,1.6039999723434448 -0.0010000000474974513,9.385000228881836 -0.0010000000474974513,9.385000228881836 C-0.0010000000474974513,9.385000228881836 7.7789998054504395,1.6039999723434448 7.7789998054504395,1.6039999723434448 C7.7789998054504395,1.6039999723434448 8.84000015258789,0.5440000295639038 8.84000015258789,0.5440000295639038 C10.98799991607666,-1.6030000448226929 10.98900032043457,-5.085999965667725 8.842000007629395,-7.234000205993652 C8.840999603271484,-7.235000133514404 8.840999603271484,-7.235000133514404 8.84000015258789,-7.236000061035156z"
                fill="rgb(0,0,0)"
                fill-opacity="1"
              />
            </g>
          </g>
          <g
            opacity="1"
            style="display: block;"
            transform="matrix(1,0,0,1,4,4)"
          >
            <g
              opacity="1"
              transform="matrix(1,0,0,1,12,11.845999717712402)"
            >
              <path
                d=" M8.84000015258789,-7.236000061035156 C6.692999839782715,-9.383999824523926 3.2090001106262207,-9.385000228881836 1.0609999895095825,-7.236999988555908 C1.0609999895095825,-7.236999988555908 1.059999942779541,-7.236000061035156 1.059000015258789,-7.236000061035156 C1.059000015258789,-7.236000061035156 -0.0010000000474974513,-6.176000118255615 -0.0010000000474974513,-6.176000118255615 C-0.0010000000474974513,-6.176000118255615 -1.0609999895095825,-7.236000061035156 -1.0609999895095825,-7.236000061035156 C-3.2090001106262207,-9.383999824523926 -6.690999984741211,-9.383999824523926 -8.84000015258789,-7.236000061035156 C-10.98799991607666,-5.0879998207092285 -10.98799991607666,-1.6039999723434448 -8.84000015258789,0.5440000295639038 C-8.84000015258789,0.5440000295639038 -7.781000137329102,1.6039999723434448 -7.781000137329102,1.6039999723434448 C-7.781000137329102,1.6039999723434448 -0.0010000000474974513,9.385000228881836 -0.0010000000474974513,9.385000228881836 C-0.0010000000474974513,9.385000228881836 7.7789998054504395,1.6039999723434448 7.7789998054504395,1.6039999723434448 C7.7789998054504395,1.6039999723434448 8.84000015258789,0.5440000295639038 8.84000015258789,0.5440000295639038 C10.98799991607666,-1.6030000448226929 10.98900032043457,-5.085999965667725 8.842000007629395,-7.234000205993652 C8.840999603271484,-7.235000133514404 8.840999603271484,-7.235000133514404 8.84000015258789,-7.236000061035156z"
                fill-opacity="0"
                stroke="rgb(0,0,0)"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-opacity="1"
                stroke-width="2"
              />
            </g>
          </g>
        </g>
      </svg>
    </div>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;

exports[`HeartButton renders without error 1`] = `
<DocumentFragment>
  <button
    aria-label="Like Post"
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
    data-mui-internal-clone-element="true"
    tabindex="0"
    type="button"
  >
    <div
      style="overflow: hidden; outline: none; width: 28px; height: 28px;"
    >
      <svg
        height="32"
        id="heart_i5001"
        preserveAspectRatio="xMidYMid meet"
        style="width: 100%; height: 100%; transform: translate3d(0,0,0);"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs>
          <clippath
            id="__lottie_element_2"
          >
            <rect
              height="32"
              width="32"
              x="0"
              y="0"
            />
          </clippath>
        </defs>
        <g
          clip-path="url(#__lottie_element_2)"
        >
          <g
            opacity="1"
            style="display: block;"
            transform="matrix(1,0,0,1,4,4)"
          >
            <g
              opacity="0"
              transform="matrix(0.5,0,0,0.5,12,11.845999717712402)"
            >
              <path
                d=" M8.84000015258789,-7.236000061035156 C6.692999839782715,-9.383999824523926 3.2090001106262207,-9.385000228881836 1.0609999895095825,-7.236999988555908 C1.0609999895095825,-7.236999988555908 1.059999942779541,-7.236000061035156 1.059000015258789,-7.236000061035156 C1.059000015258789,-7.236000061035156 -0.0010000000474974513,-6.176000118255615 -0.0010000000474974513,-6.176000118255615 C-0.0010000000474974513,-6.176000118255615 -1.0609999895095825,-7.236000061035156 -1.0609999895095825,-7.236000061035156 C-3.2090001106262207,-9.383999824523926 -6.690999984741211,-9.383999824523926 -8.84000015258789,-7.236000061035156 C-10.98799991607666,-5.0879998207092285 -10.98799991607666,-1.6039999723434448 -8.84000015258789,0.5440000295639038 C-8.84000015258789,0.5440000295639038 -7.781000137329102,1.6039999723434448 -7.781000137329102,1.6039999723434448 C-7.781000137329102,1.6039999723434448 -0.0010000000474974513,9.385000228881836 -0.0010000000474974513,9.385000228881836 C-0.0010000000474974513,9.385000228881836 7.7789998054504395,1.6039999723434448 7.7789998054504395,1.6039999723434448 C7.7789998054504395,1.6039999723434448 8.84000015258789,0.5440000295639038 8.84000015258789,0.5440000295639038 C10.98799991607666,-1.6030000448226929 10.98900032043457,-5.085999965667725 8.842000007629395,-7.234000205993652 C8.840999603271484,-7.235000133514404 8.840999603271484,-7.235000133514404 8.84000015258789,-7.236000061035156z"
                fill="rgb(0,0,0)"
                fill-opacity="1"
              />
            </g>
          </g>
          <g
            opacity="1"
            style="display: block;"
            transform="matrix(1,0,0,1,4,4)"
          >
            <g
              opacity="1"
              transform="matrix(1,0,0,1,12,11.845999717712402)"
            >
              <path
                d=" M8.84000015258789,-7.236000061035156 C6.692999839782715,-9.383999824523926 3.2090001106262207,-9.385000228881836 1.0609999895095825,-7.236999988555908 C1.0609999895095825,-7.236999988555908 1.059999942779541,-7.236000061035156 1.059000015258789,-7.236000061035156 C1.059000015258789,-7.236000061035156 -0.0010000000474974513,-6.176000118255615 -0.0010000000474974513,-6.176000118255615 C-0.0010000000474974513,-6.176000118255615 -1.0609999895095825,-7.236000061035156 -1.0609999895095825,-7.236000061035156 C-3.2090001106262207,-9.383999824523926 -6.690999984741211,-9.383999824523926 -8.84000015258789,-7.236000061035156 C-10.98799991607666,-5.0879998207092285 -10.98799991607666,-1.6039999723434448 -8.84000015258789,0.5440000295639038 C-8.84000015258789,0.5440000295639038 -7.781000137329102,1.6039999723434448 -7.781000137329102,1.6039999723434448 C-7.781000137329102,1.6039999723434448 -0.0010000000474974513,9.385000228881836 -0.0010000000474974513,9.385000228881836 C-0.0010000000474974513,9.385000228881836 7.7789998054504395,1.6039999723434448 7.7789998054504395,1.6039999723434448 C7.7789998054504395,1.6039999723434448 8.84000015258789,0.5440000295639038 8.84000015258789,0.5440000295639038 C10.98799991607666,-1.6030000448226929 10.98900032043457,-5.085999965667725 8.842000007629395,-7.234000205993652 C8.840999603271484,-7.235000133514404 8.840999603271484,-7.235000133514404 8.84000015258789,-7.236000061035156z"
                fill-opacity="0"
                stroke="rgb(0,0,0)"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-opacity="1"
                stroke-width="2"
              />
            </g>
          </g>
        </g>
      </svg>
    </div>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
